pixel-avatar-hardhat/contracts/PixelAvatars.sol

```diff
@@ -64,60 +30,59 @@ contract PixelAvatars is ERC721Enumerable, ReentrancyGuard, Ownable {
         baseURI = _newBaseURI;
     }
 
+    function setMintPrice(uint256 _newPrice) public onlyOwner {
+        // Mint price in wei
+        mintPrice = _newPrice;
+    }
+
+    function setServerAddress(address _address) public onlyOwner {
+        serverAddress = _address;
+    }
+
     event LogTokenMinted(address minter, uint256 tokenId);
 
-    /// @dev Private sale minting (reserved for DevDao owners)
-    function mintWithDevDaoToken(uint256 _devDaoTokenId)
-        public
-        payable
-        nonReentrant
-        validDevDaoToken(_devDaoTokenId)
-        devDaoTokenOwnerOf(_devDaoTokenId)
+    modifier validServerSignature(
+        uint256 tokenId,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    )
     {
-        require(mintPrice <= msg.value, "Not enough ether sent");
-        console.log(
-            "mintWithDevDaoToken | _devDaoTokenId '%s'",
-            _devDaoTokenId
+        bytes memory message = abi.encodePacked(
+            "TokenId:", Strings.toString(tokenId),
+            "Address:", msg.sender,
+            "Deadline:", Strings.toString(deadline)
         );
 
-        _safeMint(msg.sender, _devDaoTokenId);
-        emit LogTokenMinted(msg.sender, _devDaoTokenId);
-    }
+        address signer = ecrecover(keccak256(message), v, r, s);
 
-    // provide Mint price in wei
-    function setMintPrice(uint256 _newPrice) public onlyOwner {
-        mintPrice = _newPrice;
-    }
+        require(signer == serverAddress, string('Invalid server signature'));
+        require(block.timestamp <= deadline, "Signature expired");
 
-    function withdraw() public onlyOwner {
-        payable(msg.sender).transfer(address(this).balance);
+        _;
     }
 
-    function multiMintWithDevDaoToken(uint256[] memory _devDaoTokenIds)
-        public
-        payable
-        nonReentrant
-        multipleDevDaoTokenOwnerOf(_devDaoTokenIds)
+    function mintWithSignature(
```

Comment:
Function looks good, but the business logic was previously supporting minting more than one token per transaction. Looks like we can now only mint a single token per transaction and I'm not sure if that is a desired change.